### ISSUE TITLE
Compile with -parameters flag for reflection

### ISSUE THREAD

**Issue by rnett** [tensorflow/java#144]
I'm trying to go some reflection based codegen from your op creation methods (the stuff in Ops), specifically for more idiomatic kotlin methods (default args instead of Options).  However, since parameter names aren't present for reflection, making these match is currently quite a pain (I'm pulling and parsing the api_def protos).

I doubt I'm only person who will be doing codgen, this would be very helpful for anyone who tries to.

Afaik it's extremely easy to enable in maven, I'd make a PR but I don't want to touch build logic.

**Comment by Craigacp** [comment]
I'm not sure we want to do this, as it makes the parameter names part of the public API, when at the moment they are not. That said, we still don't have a stability guarantee on the API so they can freely change until we make such a guarantee.

Before turning this on at minimum we should have a pass to make sure the parameter names are all consistent across the API.

**Comment by rnett** [comment]
True, although reflection is prone to breaking when you change anything.

A probably better solution would be to provide some representation of your generated op classes and methods, like the api def protos for tensorflow.  Theoretically the tensorflow ones should be all you need but there are some cases when things don't match up (mostly just names, i.e. dtype vs T).  There is a java_api proto directory [here](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/core/api_def/java_api) but it's for the old api.  It's a whole lot more work though and seems like overkill for the relatively few people who will be doing codegen based on it.
